v0.14.2 panic: value is marked, so must be unmarked first

<!--
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

If your issue relates to a specific Terraform provider, please open it in the provider's own repository. The index of providers is at https://github.com/terraform-providers.

To fix problems, we need clear reproduction cases - we need to be able to see it happen locally. A reproduction case is ideally something a Terraform Core engineer can git-clone or copy-paste and run immediately, without inventing any details or context. 

* A short example can be directly copy-pasteable; longer examples should be in separate git repositories, especially if multiple files are needed
* Please include all needed context. For example, if you figured out that an expression can cause a crash, put the expression in a variable definition or a resource
* Set defaults on (or omit) any variables. The person reproducing it should not need to invent variable settings
* If multiple steps are required, such as running terraform twice, consider scripting it in a simple shell script. For example, see [this case](https://github.com/danieldreier/terraform-issue-reproductions/tree/master/25719). Providing a script can be easier than explaining what changes to make to the config between runs.
* Omit any unneeded complexity: remove variables, conditional statements, functions, modules, providers, and resources that are not needed to trigger the bug
* When possible, use the [null resource](https://www.terraform.io/docs/providers/null/resource.html) provider rather than a real provider in order to minimize external dependencies. We know this isn't always feasible. The Terraform Core team doesn't have deep domain knowledge in every provider, or access to every cloud platform for reproduction cases.

-->

### Terraform Version
<!---
Run `terraform version` to show the version, and paste the result between the ``` marks below.

If you are not running the latest version of Terraform, please try upgrading because your issue may have already been fixed.
-->

```
Terraform v0.14.2
```

### Terraform Configuration Files
<!--
Paste the relevant parts of your Terraform configuration between the ``` marks below.

For Terraform configs larger than a few resources, or that involve multiple files, please make a GitHub repository that we can clone, rather than copy-pasting multiple files in here. For security, you can also encrypt the files using our GPG public key at https://www.hashicorp.com/security.
-->

```terraform
variable "first" {
  type      = object({prop = string})
  sensitive = true
  default   = {prop = "first"}
}

variable "second" {
  type      = object({prop = string})
  sensitive = true
  default   = {prop = "second"}
}

locals {
  properties     = [var.first, var.second]
  no_crash_local = merge(local.properties...)

  properties_copy = [for i in range(0, length(local.properties)) : local.properties[i]]
  crash_local     = merge(local.properties_copy...)
}

```
 

### Crash Output
<!--
If the console output indicates that Terraform crashed, please share a link to a GitHub Gist containing the output of the `crash.log` file.
-->

```
2020/12/09 14:18:58 [INFO] Terraform version: 0.14.2  
2020/12/09 14:18:58 [INFO] Go runtime version: go1.15.2
2020/12/09 14:18:58 [INFO] CLI args: []string{"/usr/local/bin/terraform", "plan"}
2020/12/09 14:18:58 [DEBUG] Attempting to open CLI config file: /Users/[redacted]/.terraformrc
2020/12/09 14:18:58 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2020/12/09 14:18:58 [DEBUG] checking for credentials in "/Users/[redacted]/.terraform.d/plugins"
2020/12/09 14:18:58 [DEBUG] checking for credentials in "/Users/[redacted]/.terraform.d/plugins/darwin_amd64"
2020/12/09 14:18:58 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2020/12/09 14:18:58 [DEBUG] will search for provider plugins in /Users/[redacted]/.terraform.d/plugins
2020/12/09 14:18:58 [TRACE] getproviders.SearchLocalDirectory: /Users/[redacted]/.terraform.d/plugins is a symlink to /Users/[redacted]/.terraform.d/plugins
2020/12/09 14:18:58 [DEBUG] ignoring non-existing provider search directory /Users/[redacted]/Library/Application Support/io.terraform/plugins
2020/12/09 14:18:58 [DEBUG] ignoring non-existing provider search directory /Library/Application Support/io.terraform/plugins
2020/12/09 14:18:58 [INFO] CLI command args: []string{"plan"}
2020/12/09 14:18:58 [TRACE] Meta.Backend: no config given or present on disk, so returning nil config
2020/12/09 14:18:58 [TRACE] Meta.Backend: backend has not previously been initialized in this working directory
2020/12/09 14:18:58 [DEBUG] New state was assigned lineage "5ec85270-6b10-5d54-24f5-c50a8a0f90d2"
2020/12/09 14:18:58 [TRACE] Meta.Backend: using default local state only (no backend configuration, and no existing initialized backend)
2020/12/09 14:18:58 [TRACE] Meta.Backend: instantiated backend of type <nil>
2020/12/09 14:18:58 [TRACE] providercache.fillMetaCache: scanning directory .terraform/providers
2020/12/09 14:18:58 [TRACE] getproviders.SearchLocalDirectory: .terraform/providers is a symlink to .terraform/providers
2020/12/09 14:18:58 [TRACE] getproviders.SearchLocalDirectory: found registry.terraform.io/hashicorp/aws v3.20.0 for darwin_amd64 at .terraform/providers/registry.terraform.io/hashicorp/aws/3.20.0/darwin_amd64
2020/12/09 14:18:58 [TRACE] getproviders.SearchLocalDirectory: found registry.terraform.io/hashicorp/local v2.0.0 for darwin_amd64 at .terraform/providers/registry.terraform.io/hashicorp/local/2.0.0/darwin_amd64
2020/12/09 14:18:58 [TRACE] getproviders.SearchLocalDirectory: found registry.terraform.io/hashicorp/local v2.0.0 for linux_amd64 at .terraform/providers/registry.terraform.io/hashicorp/local/2.0.0/linux_amd64
2020/12/09 14:18:58 [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.terraform.io/hashicorp/aws/3.20.0/darwin_amd64 as a candidate package for registry.terraform.io/hashicorp/aws 3.20.0
2020/12/09 14:18:58 [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.terraform.io/hashicorp/local/2.0.0/darwin_amd64 as a candidate package for registry.terraform.io/hashicorp/local 2.0.0
2020/12/09 14:18:58 [TRACE] providercache.fillMetaCache: ignoring .terraform/providers/registry.terraform.io/hashicorp/local/2.0.0/linux_amd64 because it is for linux_amd64, not darwin_amd64
2020/12/09 14:18:59 [TRACE] providercache.fillMetaCache: using cached result from previous scan of .terraform/providers
2020/12/09 14:18:59 [DEBUG] checking for provisioner in "."
2020/12/09 14:18:59 [DEBUG] checking for provisioner in "/usr/local/bin"
2020/12/09 14:18:59 [DEBUG] checking for provisioner in "/Users/[redacted]/.terraform.d/plugins"
2020/12/09 14:18:59 [DEBUG] checking for provisioner in "/Users/[redacted]/.terraform.d/plugins/darwin_amd64"
2020/12/09 14:18:59 [INFO] Failed to read plugin lock file .terraform/plugins/darwin_amd64/lock.json: open .terraform/plugins/darwin_amd64/lock.json: no such file or directory
2020/12/09 14:18:59 [TRACE] Meta.Backend: backend <nil> does not support operations, so wrapping it in a local backend
2020/12/09 14:18:59 [INFO] backend/local: starting Plan operation
2020/12/09 14:18:59 [TRACE] backend/local: requesting state manager for workspace "default"
2020/12/09 14:18:59 [TRACE] backend/local: state manager for workspace "default" will:
 - read initial snapshot from terraform.tfstate
 - write new snapshots to terraform.tfstate
 - create any backup at terraform.tfstate.backup
2020/12/09 14:18:59 [TRACE] backend/local: requesting state lock for workspace "default"
2020/12/09 14:18:59 [TRACE] statemgr.Filesystem: preparing to manage state snapshots at terraform.tfstate
2020/12/09 14:18:59 [TRACE] statemgr.Filesystem: existing snapshot has lineage "7605d3ec-7fb8-0f8a-1e1b-9f99606ea2f2" serial 1
2020/12/09 14:18:59 [TRACE] statemgr.Filesystem: locking terraform.tfstate using fcntl flock
2020/12/09 14:18:59 [TRACE] statemgr.Filesystem: writing lock metadata to .terraform.tfstate.lock.info
2020/12/09 14:18:59 [TRACE] backend/local: reading remote state for workspace "default"
2020/12/09 14:18:59 [TRACE] statemgr.Filesystem: reading latest snapshot from terraform.tfstate
2020/12/09 14:18:59 [TRACE] statemgr.Filesystem: read snapshot with lineage "7605d3ec-7fb8-0f8a-1e1b-9f99606ea2f2" serial 1
2020/12/09 14:18:59 [TRACE] backend/local: retrieving local state snapshot for workspace "default"
2020/12/09 14:18:59 [TRACE] backend/local: building context for current working directory
2020/12/09 14:18:59 [TRACE] terraform.NewContext: starting
2020/12/09 14:18:59 [TRACE] terraform.NewContext: loading provider schemas
2020/12/09 14:18:59 [TRACE] terraform.NewContext: complete
2020/12/09 14:18:59 [TRACE] backend/local: finished building terraform.Context
2020/12/09 14:18:59 [TRACE] backend/local: requesting interactive input, if necessary
2020/12/09 14:18:59 [TRACE] Context.Input: Prompting for provider arguments
2020/12/09 14:18:59 [TRACE] backend/local: running validation operation
2020/12/09 14:18:59 [INFO] terraform: building graph: GraphTypeValidate
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.ConfigTransformer
2020/12/09 14:18:59 [TRACE] ConfigTransformer: Starting for path: 
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.ConfigTransformer (no changes)
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.RootVariableTransformer
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.RootVariableTransformer with new graph:
  var.first - *terraform.NodeRootVariable
  var.second - *terraform.NodeRootVariable
  ------
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.ModuleVariableTransformer
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.ModuleVariableTransformer (no changes)
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.LocalTransformer
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.LocalTransformer with new graph:
  local.crash_local (expand) - *terraform.nodeExpandLocal
  local.no_crash_local (expand) - *terraform.nodeExpandLocal
  local.properties (expand) - *terraform.nodeExpandLocal
  local.properties_copy (expand) - *terraform.nodeExpandLocal
  var.first - *terraform.NodeRootVariable
  var.second - *terraform.NodeRootVariable
  ------
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.OutputTransformer
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.OutputTransformer (no changes)
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.OrphanResourceInstanceTransformer
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.OrphanResourceInstanceTransformer (no changes)
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.StateTransformer
2020/12/09 14:18:59 [TRACE] StateTransformer: state is empty, so nothing to do
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.StateTransformer (no changes)
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.AttachStateTransformer
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.AttachStateTransformer (no changes)
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.OrphanOutputTransformer
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.OrphanOutputTransformer (no changes)
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.AttachResourceConfigTransformer
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.AttachResourceConfigTransformer (no changes)
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.MissingProvisionerTransformer
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.MissingProvisionerTransformer (no changes)
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.ProvisionerTransformer
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.ProvisionerTransformer (no changes)
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.graphTransformerMulti
2020/12/09 14:18:59 [TRACE] (graphTransformerMulti) Executing graph transform *terraform.ProviderConfigTransformer
2020/12/09 14:18:59 [TRACE] (graphTransformerMulti) Completed graph transform *terraform.ProviderConfigTransformer with new graph:
  local.crash_local (expand) - *terraform.nodeExpandLocal
  local.no_crash_local (expand) - *terraform.nodeExpandLocal
  local.properties (expand) - *terraform.nodeExpandLocal
  local.properties_copy (expand) - *terraform.nodeExpandLocal
  var.first - *terraform.NodeRootVariable
  var.second - *terraform.NodeRootVariable
  ------
2020/12/09 14:18:59 [TRACE] (graphTransformerMulti) Executing graph transform *terraform.MissingProviderTransformer
2020/12/09 14:18:59 [TRACE] (graphTransformerMulti) Completed graph transform *terraform.MissingProviderTransformer (no changes)
2020/12/09 14:18:59 [TRACE] (graphTransformerMulti) Executing graph transform *terraform.ProviderTransformer
2020/12/09 14:18:59 [TRACE] (graphTransformerMulti) Completed graph transform *terraform.ProviderTransformer (no changes)
2020/12/09 14:18:59 [TRACE] (graphTransformerMulti) Executing graph transform *terraform.PruneProviderTransformer
2020/12/09 14:18:59 [TRACE] (graphTransformerMulti) Completed graph transform *terraform.PruneProviderTransformer (no changes)
2020/12/09 14:18:59 [TRACE] (graphTransformerMulti) Executing graph transform *terraform.ParentProviderTransformer
2020/12/09 14:18:59 [TRACE] (graphTransformerMulti) Completed graph transform *terraform.ParentProviderTransformer (no changes)
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.graphTransformerMulti (no changes)
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.RemovedModuleTransformer
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.RemovedModuleTransformer (no changes)
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.AttachSchemaTransformer
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.AttachSchemaTransformer (no changes)
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.ModuleExpansionTransformer
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.ModuleExpansionTransformer (no changes)
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.ReferenceTransformer
2020/12/09 14:18:59 [DEBUG] ReferenceTransformer: "var.first" references: []
2020/12/09 14:18:59 [DEBUG] ReferenceTransformer: "var.second" references: []
2020/12/09 14:18:59 [DEBUG] ReferenceTransformer: "local.no_crash_local (expand)" references: [local.properties (expand)]
2020/12/09 14:18:59 [DEBUG] ReferenceTransformer: "local.properties_copy (expand)" references: [local.properties (expand) local.properties (expand)]
2020/12/09 14:18:59 [DEBUG] ReferenceTransformer: "local.crash_local (expand)" references: [local.properties_copy (expand)]
2020/12/09 14:18:59 [DEBUG] ReferenceTransformer: "local.properties (expand)" references: [var.first var.second]
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.ReferenceTransformer with new graph:
  local.crash_local (expand) - *terraform.nodeExpandLocal
    local.properties_copy (expand) - *terraform.nodeExpandLocal
  local.no_crash_local (expand) - *terraform.nodeExpandLocal
    local.properties (expand) - *terraform.nodeExpandLocal
  local.properties (expand) - *terraform.nodeExpandLocal
    var.first - *terraform.NodeRootVariable
    var.second - *terraform.NodeRootVariable
  local.properties_copy (expand) - *terraform.nodeExpandLocal
    local.properties (expand) - *terraform.nodeExpandLocal
  var.first - *terraform.NodeRootVariable
  var.second - *terraform.NodeRootVariable
  ------
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.AttachDependenciesTransformer
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.AttachDependenciesTransformer (no changes)
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.attachDataResourceDependenciesTransformer
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.attachDataResourceDependenciesTransformer (no changes)
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.TargetsTransformer
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.TargetsTransformer (no changes)
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.ForcedCBDTransformer
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.ForcedCBDTransformer (no changes)
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.CountBoundaryTransformer
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.CountBoundaryTransformer with new graph:
  local.crash_local (expand) - *terraform.nodeExpandLocal
    local.properties_copy (expand) - *terraform.nodeExpandLocal
  local.no_crash_local (expand) - *terraform.nodeExpandLocal
    local.properties (expand) - *terraform.nodeExpandLocal
  local.properties (expand) - *terraform.nodeExpandLocal
    var.first - *terraform.NodeRootVariable
    var.second - *terraform.NodeRootVariable
  local.properties_copy (expand) - *terraform.nodeExpandLocal
    local.properties (expand) - *terraform.nodeExpandLocal
  meta.count-boundary (EachMode fixup) - *terraform.NodeCountBoundary
    local.crash_local (expand) - *terraform.nodeExpandLocal
    local.no_crash_local (expand) - *terraform.nodeExpandLocal
    local.properties (expand) - *terraform.nodeExpandLocal
    local.properties_copy (expand) - *terraform.nodeExpandLocal
    var.first - *terraform.NodeRootVariable
    var.second - *terraform.NodeRootVariable
  var.first - *terraform.NodeRootVariable
  var.second - *terraform.NodeRootVariable
  ------
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.CloseProviderTransformer
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.CloseProviderTransformer (no changes)
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.CloseRootModuleTransformer
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.CloseRootModuleTransformer with new graph:
  local.crash_local (expand) - *terraform.nodeExpandLocal
    local.properties_copy (expand) - *terraform.nodeExpandLocal
  local.no_crash_local (expand) - *terraform.nodeExpandLocal
    local.properties (expand) - *terraform.nodeExpandLocal
  local.properties (expand) - *terraform.nodeExpandLocal
    var.first - *terraform.NodeRootVariable
    var.second - *terraform.NodeRootVariable
  local.properties_copy (expand) - *terraform.nodeExpandLocal
    local.properties (expand) - *terraform.nodeExpandLocal
  meta.count-boundary (EachMode fixup) - *terraform.NodeCountBoundary
    local.crash_local (expand) - *terraform.nodeExpandLocal
    local.no_crash_local (expand) - *terraform.nodeExpandLocal
    local.properties (expand) - *terraform.nodeExpandLocal
    local.properties_copy (expand) - *terraform.nodeExpandLocal
    var.first - *terraform.NodeRootVariable
    var.second - *terraform.NodeRootVariable
  root - *terraform.nodeCloseModule
    meta.count-boundary (EachMode fixup) - *terraform.NodeCountBoundary
  var.first - *terraform.NodeRootVariable
  var.second - *terraform.NodeRootVariable
  ------
2020/12/09 14:18:59 [TRACE] Executing graph transform *terraform.TransitiveReductionTransformer
2020/12/09 14:18:59 [TRACE] Completed graph transform *terraform.TransitiveReductionTransformer with new graph:
  local.crash_local (expand) - *terraform.nodeExpandLocal
    local.properties_copy (expand) - *terraform.nodeExpandLocal
  local.no_crash_local (expand) - *terraform.nodeExpandLocal
    local.properties (expand) - *terraform.nodeExpandLocal
  local.properties (expand) - *terraform.nodeExpandLocal
    var.first - *terraform.NodeRootVariable
    var.second - *terraform.NodeRootVariable
  local.properties_copy (expand) - *terraform.nodeExpandLocal
    local.properties (expand) - *terraform.nodeExpandLocal
  meta.count-boundary (EachMode fixup) - *terraform.NodeCountBoundary
    local.crash_local (expand) - *terraform.nodeExpandLocal
    local.no_crash_local (expand) - *terraform.nodeExpandLocal
  root - *terraform.nodeCloseModule
    meta.count-boundary (EachMode fixup) - *terraform.NodeCountBoundary
  var.first - *terraform.NodeRootVariable
  var.second - *terraform.NodeRootVariable
  ------
2020/12/09 14:18:59 [DEBUG] Starting graph walk: walkValidate
2020/12/09 14:18:59 [TRACE] dag/walk: visiting "var.second"
2020/12/09 14:18:59 [TRACE] dag/walk: visiting "var.first"
2020/12/09 14:18:59 [TRACE] vertex "var.second": starting visit (*terraform.NodeRootVariable)
2020/12/09 14:18:59 [TRACE] vertex "var.first": starting visit (*terraform.NodeRootVariable)
2020/12/09 14:18:59 [TRACE] vertex "var.second": visit complete
2020/12/09 14:18:59 [TRACE] vertex "var.first": visit complete
2020/12/09 14:18:59 [TRACE] dag/walk: visiting "local.properties (expand)"
2020/12/09 14:18:59 [TRACE] vertex "local.properties (expand)": starting visit (*terraform.nodeExpandLocal)
2020/12/09 14:18:59 [TRACE] vertex "local.properties (expand)": expanding dynamic subgraph
2020/12/09 14:18:59 [TRACE] Expanding local: adding local.properties as *terraform.NodeLocal
2020/12/09 14:18:59 [TRACE] vertex "local.properties (expand)": entering dynamic subgraph
2020/12/09 14:18:59 [TRACE] dag/walk: visiting "local.properties"
2020/12/09 14:18:59 [TRACE] vertex "local.properties": starting visit (*terraform.NodeLocal)
2020/12/09 14:18:59 [TRACE] vertex "local.properties": visit complete
2020/12/09 14:18:59 [TRACE] vertex "local.properties (expand)": dynamic subgraph completed successfully
2020/12/09 14:18:59 [TRACE] vertex "local.properties (expand)": visit complete
2020/12/09 14:18:59 [TRACE] dag/walk: visiting "local.no_crash_local (expand)"
2020/12/09 14:18:59 [TRACE] vertex "local.no_crash_local (expand)": starting visit (*terraform.nodeExpandLocal)
2020/12/09 14:18:59 [TRACE] vertex "local.no_crash_local (expand)": expanding dynamic subgraph
2020/12/09 14:18:59 [TRACE] Expanding local: adding local.no_crash_local as *terraform.NodeLocal
2020/12/09 14:18:59 [TRACE] vertex "local.no_crash_local (expand)": entering dynamic subgraph
2020/12/09 14:18:59 [TRACE] dag/walk: visiting "local.no_crash_local"
2020/12/09 14:18:59 [TRACE] vertex "local.no_crash_local": starting visit (*terraform.NodeLocal)
2020/12/09 14:18:59 [TRACE] dag/walk: visiting "local.properties_copy (expand)"
2020/12/09 14:18:59 [TRACE] vertex "local.properties_copy (expand)": starting visit (*terraform.nodeExpandLocal)
2020/12/09 14:18:59 [TRACE] vertex "local.properties_copy (expand)": expanding dynamic subgraph
2020/12/09 14:18:59 [TRACE] Expanding local: adding local.properties_copy as *terraform.NodeLocal
2020/12/09 14:18:59 [TRACE] vertex "local.properties_copy (expand)": entering dynamic subgraph
2020/12/09 14:18:59 [TRACE] dag/walk: visiting "local.properties_copy"
2020/12/09 14:18:59 [TRACE] vertex "local.properties_copy": starting visit (*terraform.NodeLocal)
2020/12/09 14:18:59 [TRACE] vertex "local.no_crash_local": visit complete
2020/12/09 14:18:59 [TRACE] vertex "local.no_crash_local (expand)": dynamic subgraph completed successfully
2020/12/09 14:18:59 [TRACE] vertex "local.no_crash_local (expand)": visit complete
2020/12/09 14:18:59 [TRACE] vertex "local.properties_copy": visit complete
2020/12/09 14:18:59 [TRACE] vertex "local.properties_copy (expand)": dynamic subgraph completed successfully
2020/12/09 14:18:59 [TRACE] vertex "local.properties_copy (expand)": visit complete
2020/12/09 14:18:59 [TRACE] dag/walk: visiting "local.crash_local (expand)"
2020/12/09 14:18:59 [TRACE] vertex "local.crash_local (expand)": starting visit (*terraform.nodeExpandLocal)
2020/12/09 14:18:59 [TRACE] vertex "local.crash_local (expand)": expanding dynamic subgraph
2020/12/09 14:18:59 [TRACE] Expanding local: adding local.crash_local as *terraform.NodeLocal
2020/12/09 14:18:59 [TRACE] vertex "local.crash_local (expand)": entering dynamic subgraph
2020/12/09 14:18:59 [TRACE] dag/walk: visiting "local.crash_local"
2020/12/09 14:18:59 [TRACE] vertex "local.crash_local": starting visit (*terraform.NodeLocal)
2020/12/09 14:18:59 [TRACE] vertex "local.crash_local": visit complete
panic: value is marked, so must be unmarked first

goroutine 98 [running]:
github.com/zclconf/go-cty/cty.Value.assertUnmarked(...)
	/go/pkg/mod/github.com/zclconf/go-cty@v1.7.0/cty/marks.go:123
github.com/zclconf/go-cty/cty.Value.LengthInt(0x384f040, 0xc0006f6ce0, 0x3145580, 0xc0006f6d20, 0x1)
	/go/pkg/mod/github.com/zclconf/go-cty@v1.7.0/cty/value_ops.go:1025 +0x4f
github.com/hashicorp/hcl/v2/hclsyntax.(*FunctionCallExpr).Value(0xc0005004b0, 0xc0006f7300, 0x0, 0xc000957800, 0x1, 0x1, 0x0, 0x0, 0x0)
	/go/pkg/mod/github.com/hashicorp/hcl/v2@v2.8.0/hclsyntax/expression.go:320 +0x3ba5
github.com/hashicorp/terraform/lang.(*Scope).EvalExpr(0xc0006f8410, 0x384da40, 0xc0005004b0, 0x384ef00, 0x49372f0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/circleci/project/project/lang/eval.go:171 +0x1b7
github.com/hashicorp/terraform/terraform.(*BuiltinEvalContext).EvaluateExpr(0xc000948270, 0x384da40, 0xc0005004b0, 0x384ef00, 0x49372f0, 0x0, 0x0, 0x0, 0x106d060, 0xc000679a20, ...)
	/home/circleci/project/project/terraform/eval_context_builtin.go:287 +0xbb
github.com/hashicorp/terraform/terraform.(*NodeLocal).Execute(0xc000a894a0, 0x388a660, 0xc000948270, 0xc00003a004, 0x309dce0, 0x3223be0)
	/home/circleci/project/project/terraform/node_local.go:156 +0x71d
github.com/hashicorp/terraform/terraform.(*ContextGraphWalker).Execute(0xc0008bb450, 0x388a660, 0xc000948270, 0x2ce0d9f8, 0xc000a894a0, 0x0, 0x0, 0x0)
	/home/circleci/project/project/terraform/graph_walk_context.go:127 +0xbc
github.com/hashicorp/terraform/terraform.(*Graph).walk.func1(0x3223be0, 0xc000a894a0, 0x0, 0x0, 0x0)
	/home/circleci/project/project/terraform/graph.go:59 +0x962
github.com/hashicorp/terraform/dag.(*Walker).walkVertex(0xc0006f41e0, 0x3223be0, 0xc000a894a0, 0xc000a982c0)
	/home/circleci/project/project/dag/walk.go:387 +0x375
created by github.com/hashicorp/terraform/dag.(*Walker).Update
	/home/circleci/project/project/dag/walk.go:309 +0x1246

```

### Expected Behavior
<!--
What should have happened?
-->
Command success

### Actual Behavior
<!--
What actually happened?
-->
Panic / terraform crash

### Steps to Reproduce
<!--
Please list the full steps required to reproduce the issue, for example:
1. `terraform init`
2. `terraform apply`
-->
```
$ terraform init
$ terraform plan
```

### Additional Context
<!--
Are there anything atypical about your situation that we should know? For example: is Terraform running in a wrapper script or in a CI system? Are you passing any unusual command line options or environment variables to opt-in to non-default behavior?
-->

### References
<!--
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:

- #6017

-->

seems related to:

https://github.com/hashicorp/terraform/pull/27187

https://github.com/hashicorp/terraform/issues/27107
https://github.com/hashicorp/terraform/issues/27040

--- a/go.mod
+++ b/go.mod
@@ -114,7 +114,7 @@ require (
 	github.com/xanzy/ssh-agent v0.2.1
 	github.com/xiang90/probing v0.0.0-20160813154853-07dd2e8dfe18 // indirect
 	github.com/xlab/treeprint v0.0.0-20161029104018-1d6e34225557
-	github.com/zclconf/go-cty v1.7.0
+	github.com/zclconf/go-cty v1.7.1
 	github.com/zclconf/go-cty-yaml v1.0.2
 	go.uber.org/atomic v1.3.2 // indirect
 	go.uber.org/multierr v1.1.0 // indirect

--- a/go.mod
+++ b/go.mod
@@ -64,7 +64,7 @@ require (
 	github.com/hashicorp/go-uuid v1.0.1
 	github.com/hashicorp/go-version v1.2.0
 	github.com/hashicorp/hcl v0.0.0-20170504190234-a4b07c25de5f
-	github.com/hashicorp/hcl/v2 v2.8.0
+	github.com/hashicorp/hcl/v2 v2.8.1
 	github.com/hashicorp/memberlist v0.1.0 // indirect
 	github.com/hashicorp/serf v0.0.0-20160124182025-e4ec8cc423bb // indirect
 	github.com/hashicorp/terraform-config-inspect v0.0.0-20191212124732-c6ae6269b9d7

